stock_restrict_lot/models/stock_move.py

```diff
@@ -92,3 +93,26 @@ def _split(self, qty, restrict_partner_id=False):
         if vals_list and self.restrict_lot_id:
             vals_list[0]["restrict_lot_id"] = self.restrict_lot_id.id
         return vals_list
+
+    def _action_done(self, cancel_backorder=False):
+        res = super()._action_done(cancel_backorder=cancel_backorder)
+        self._check_lot_consistent_with_restriction()
+        return res
+
+    def _check_lot_consistent_with_restriction(self):
+        """
+        Check that the lot set on move lines
+        is the same as the restricted lot set on the move
+        """
+        for move in self.filtered(lambda x: x.restrict_lot_id and x.move_line_ids):
+            move_line_lot = move.mapped("move_line_ids.lot_id")
+            if move.restrict_lot_id != move_line_lot:
+                raise UserError(
+                    _(
+                        "The lot(s) %(move_line_lot)s being moved is "
+                        "inconsistent with the restriction on "
+                        "lot %(move_restrict_lot)s set on the move",
+                        move_line_lot=", ".join(x.display_name for x in move_line_lot),
```

Comment:
Here you can use `mapped` function without cycle:
```py
move_line_lot=", ".join(move_line_lot.mapped('display_name')),
```
I think this code looks more clean.